AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/smithygenerated/aws_cryptography_keystore/client.py

```diff
@@ -0,0 +1,462 @@
+                    config,
+                    operation_name,
+                )
+
+                # We perform this type-ignored re-assignment because `context` needs
+                # to point at the latest context so it can be generically handled
+                # later on. This is only an issue here because we've created a copy,
+                # so we're no longer simply pointing at the same object in memory
+                # with different names and type hints. It is possible to address this
+                # without having to fall back to the type ignore, but it would impose
+                # unnecessary runtime costs.
+                context = context_with_response  # type: ignore
+
+                if isinstance(context_with_response.response, Exception):
+                    # Step 7u: Reacquire retry token if the attempt failed
+                    try:
+                        retry_token = retry_strategy.refresh_retry_token_for_retry(
+                            token_to_renew=retry_token,
+                            error_info=RetryErrorInfo(
+                                # TODO: Determine the error type.
+                                error_type=RetryErrorType.CLIENT_ERROR,
+                            ),
+                        )
+                    except SmithyRetryException:
+                        raise context_with_response.response
+                    await sleep(retry_token.retry_delay)
+                else:
+                    # Step 8: Invoke record_success
+                    retry_strategy.record_success(token=retry_token)
+                    break
+        except Exception as e:
+            context._response = e
+
+        # At this point, the context's request will have been definitively set, and
+        # The response will be set either with the modeled output or an exception. The
+        # transport_request and transport_response may be set or None.
+        execution_context = cast(
+            InterceptorContext[
+                Input, Output, DafnyRequest | None, DafnyResponse | None
+            ],
+            context,
+        )
+        return await self._finalize_execution(interceptors, execution_context)
+
+    async def _handle_attempt(
+        self,
+        deserialize: Callable[[DafnyResponse, Config], Awaitable[Output]],
+        interceptors: list[Interceptor[Input, Output, DafnyRequest, DafnyResponse]],
+        context: InterceptorContext[Input, None, DafnyRequest, None],
+        config: Config,
+        operation_name: str,
+    ) -> InterceptorContext[Input, Output, DafnyRequest, DafnyResponse | None]:
+        try:
+            # Step 7a: Invoke read_before_attempt
+            for interceptor in interceptors:
+                interceptor.read_before_attempt(context)
+
+            # Step 7m: Involve client Dafny impl
+            if config.dafnyImplInterface.impl is None:
+                raise Exception("No impl found on the operation config.")
+
```

Comment:
Should we move this up the chain?

Reply:
We could, but there's no strong need to do this. This is a weird edge case that customers should never see; as long as this happens before it's called, it should be fine

Reply:
Is it being checked somewhere? I might have missed it.